src/hal/time.cpp

```diff
@@ -39,70 +36,169 @@ void OswHal::getUTCTime(uint32_t *hour, uint32_t *minute, uint32_t *second) {
-    } else if (d.Hour() == 0) {
-      *hour = 12;
+    d.InitWithEpoch32Time(this->getTime(timezone));
+    if (!OswConfigAllKeys::timeFormat.get()) {
+      if (d.Hour() > 12) {
+        *hour = d.Hour() - 12;
+        if (afterNoon != nullptr) *afterNoon = true;
+      } else if (d.Hour() == 0) {
+        *hour = 12;
+        if (afterNoon != nullptr) *afterNoon = false;
+      } else if (d.Hour() == 12) {
+        *hour = d.Hour();
+        if (afterNoon != nullptr) *afterNoon = true;
+      } else {
+        *hour = d.Hour();
+        if (afterNoon != nullptr) *afterNoon = false;
+      }
     } else {
       *hour = d.Hour();
+      if (afterNoon != nullptr) *afterNoon = false;
     }
-  } else {
-    *hour = d.Hour();
-  }
-  *minute = d.Minute();
-  *second = d.Second();
+    *minute = d.Minute();
+    *second = d.Second();
 }
 
-void OswHal::getLocalTime(uint32_t *hour, uint32_t *minute, uint32_t *second, bool *afterNoon) {
-  RtcDateTime d = RtcDateTime();
-  d.InitWithEpoch32Time(this->getLocalTime());
-  if (!OswConfigAllKeys::timeFormat.get()) {
-    if (d.Hour() > 12) {
-      *hour = d.Hour() - 12;
-      *afterNoon = true;
-    } else if (d.Hour() == 0) {
-      *hour = 12;
-      *afterNoon = false;
-    } else if (d.Hour() == 12) {
-      *hour = d.Hour();
-      *afterNoon = true;
-    } else {
-      *hour = d.Hour();
-      *afterNoon = false;
-    }
-  } else {
-    *hour = d.Hour();
-    *afterNoon = false;
-  }
-  *minute = d.Minute();
-  *second = d.Second();
+uint32_t OswHal::getTime(short timezone) {
+  return this->getUTCTime() + (3600 * timezone) + (long)(3600 * OswConfigAllKeys::daylightOffset.get());
 }
 
-void OswHal::getDate(uint32_t *day, uint32_t *weekDay) {
+void OswHal::getDate(short timezone, uint32_t* day, uint32_t* weekDay) {
   RtcDateTime d = RtcDateTime();
-  d.InitWithEpoch32Time(this->getLocalTime());
+  d.InitWithEpoch32Time(this->getTime(timezone));
   *weekDay = d.DayOfWeek();
   *day = d.Day();
 }
 
-void OswHal::getDate(uint32_t *day, uint32_t *month, uint32_t *year) {
+void OswHal::getDate(short timezone, uint32_t* day, uint32_t* month, uint32_t* year) {
   RtcDateTime d = RtcDateTime();
-  d.InitWithEpoch32Time(getLocalTime());
+  d.InitWithEpoch32Time(this->getTime(timezone));
   *day = d.Day();
   *month = d.Month();
   *year = d.Year();
 }
 
-const char *OswHal::getWeekday() {
+const char* OswHal::getWeekday(short timezone) {
   uint32_t day = 0;
   uint32_t wDay = 0;
-  this->getDate(&day, &wDay);
+  this->getDate(timezone, &day, &wDay);
 
   const char *dayMap[7] = {LANG_SUNDAY,   LANG_MONDAY, LANG_TUESDAY, LANG_WEDNESDAY,
-                         LANG_THURSDAY, LANG_FRIDAY, LANG_SATURDAY};
+                            LANG_THURSDAY, LANG_FRIDAY, LANG_SATURDAY};
   return dayMap[wDay];
-}
\ No newline at end of file
+}
+
+
+/*
+
+uint32_t OswHal::getLocalTime() {
+  return this->getUTCTime() + 3600 * OswConfigAllKeys::timeZone.get() + (long)(3600 * OswConfigAllKeys::daylightOffset.get());
```

Comment:
Derived from merge for time-zone function, it's an older version, which is dead code.

Reply:
Oh, I thought you already removed that commented code - please, remove it!